File: Java/Floyd_Warshall_algorithm.java
import java.util.Arrays;

public class FloydWarshallAlgorithm {
    static final int INF = Integer.MAX_VALUE; // Infinity represents absence of edge between vertices

    // Function to find all-pairs shortest paths using Floyd-Warshall algorithm
    public static void floydWarshall(int[][] graph) {
        int V = graph.length;

        // Create a 2D array dist[][] to store shortest distances between every pair of vertices
        int[][] dist = new int[V][V];

        // Initialize dist[][] with the values of the input graph
        for (int i = 0; i < V; i++)
            System.arraycopy(graph[i], 0, dist[i], 0, V);

        // Iterate through all vertices and calculate shortest paths
        for (int k = 0; k < V; k++) {
            // Pick all vertices as source one by one
            for (int i = 0; i < V; i++) {
                // Pick all vertices as destination for the above picked source
                for (int j = 0; j < V; j++) {
                    // If vertex k is on the shortest path from i to j,
                    // then update the value of dist[i][j]
                    if (dist[i][k] != INF && dist[k][j] != INF &&
                            dist[i][k] + dist[k][j] < dist[i][j]) {
                        dist[i][j] = dist[i][k] + dist[k][j];
                    }
                }
            }
        }

        // Print the shortest distances
        printSolution(dist);
    }

    // Function to print the solution (shortest distances)
    public static void printSolution(int[][] dist) {
        int V = dist.length;
        System.out.println("Shortest distances between every pair of vertices:");
        for (int i = 0; i < V; ++i) {
            for (int j = 0; j < V; ++j) {
                if (dist[i][j] == INF)
                    System.out.print("INF ");
                else
                    System.out.print(dist[i][j] + "  ");
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        // Example graph represented as an adjacency matrix
        int[][] graph = {
                {0, 5, INF, 10},
                {INF, 0, 3, INF},
                {INF, INF, 0, 1},
                {INF, INF, INF, 0}
        };

        // Find all-pairs shortest paths
        floydWarshall(graph);
    }
}

File: C++/Floyd_Warshall_algorithm.cpp
#include <iostream>
#include <vector>

using namespace std;

const int INF = 1e9; // Define infinity as a very large value

void floydWarshall(vector<vector<int>>& graph) {
    int V = graph.size();

    // Initialize distances matrix
    vector<vector<int>> dist(V, vector<int>(V, INF));

    // Initialize distances for direct edges
    for (int i = 0; i < V; ++i) {
        for (int j = 0; j < V; ++j) {
            if (i == j) {
                dist[i][j] = 0; // Distance to itself is 0
            } else if (graph[i][j] != INF) {
                dist[i][j] = graph[i][j];
            }
        }
    }

    // Floyd-Warshall algorithm
    for (int k = 0; k < V; ++k) {
        for (int i = 0; i < V; ++i) {
            for (int j = 0; j < V; ++j) {
                if (dist[i][k] != INF && dist[k][j] != INF && dist[i][k] + dist[k][j] < dist[i][j]) {
                    dist[i][j] = dist[i][k] + dist[k][j];
                }
            }
        }
    }

    // Output the shortest distances
    cout << "Shortest distances between all pairs of vertices:" << endl;
    for (int i = 0; i < V; ++i) {
        for (int j = 0; j < V; ++j) {
            if (dist[i][j] == INF) {
                cout << "INF\t";
            } else {
                cout << dist[i][j] << "\t";
            }
        }
        cout << endl;
    }
}

int main() {
    // Example graph represented by adjacency matrix
    vector<vector<int>> graph = {
        {0, 5, INF, 10},
        {INF, 0, 3, INF},
        {INF, INF, 0, 1},
        {INF, INF, INF, 0}
    };

    // Run Floyd-Warshall algorithm
    floydWarshall(graph);

    return 0;
}
